src/compiler/output-targets/dist-custom-elements/index.ts

```diff
@@ -233,49 +231,76 @@ export const addCustomElementInputs = (
     bundleOpts.loader![coreKey] = exp.join('\n');
   });
 
-  if (outputTarget.customElementsExportBehavior === 'bundle') {
-    bundleOpts.loader!['\0core'] += indexImports.join('\n');
-  }
-
-  // Only re-export component definitions if the barrel export behavior is set
-  if (outputTarget.customElementsExportBehavior === 'single-export-module') {
-    bundleOpts.loader!['\0core'] += indexExports.join('\n');
-  }
-
-  if (outputTarget.customElementsExportBehavior === 'bundle') {
-    bundleOpts.loader!['\0core'] += `
-export const defineCustomElements = (opts) => {
-    if (typeof customElements !== 'undefined') {
-        [
-            ${exportNames.join(',\n          ')}
-        ].forEach(cmp => {
-            if (!customElements.get(cmp.is)) {
-                customElements.define(cmp.is, cmp, opts);
-            }
-        });
-    }
-};`;
-  }
+  // Generate the contents of the entry file to be created by the bundler
+  bundleOpts.loader!['\0core'] = generateEntryPoint(outputTarget, indexImports, indexExports, exportNames);
 };
 
 /**
  * Generate the entrypoint (`index.ts` file) contents for the `dist-custom-elements` output target
  * @param outputTarget the output target's configuration
+ * @param cmpImports The import declarations for local component modules.
+ * @param cmpExports The export declarations for local component modules.
+ * @param cmpNames The exported component names (could be aliased) from local component modules.
  * @returns the stringified contents to be placed in the entrypoint
  */
-export const generateEntryPoint = (outputTarget: d.OutputTargetDistCustomElements): string => {
-  const imp: string[] = [];
+export const generateEntryPoint = (
+  outputTarget: d.OutputTargetDistCustomElements,
+  cmpImports: string[] = [],
+  cmpExports: string[] = [],
+  cmpNames: string[] = []
+): string => {
+  const body: string[] = [];
+  const imports: string[] = [];
+  const exports: string[] = [];
 
-  imp.push(
+  // Exports that are always present
+  exports.push(
     `export { setAssetPath, setPlatformOptions } from '${STENCIL_INTERNAL_CLIENT_ID}';`,
     `export * from '${USER_INDEX_ENTRY_ID}';`
   );
 
+  // Content related to global scripts
   if (outputTarget.includeGlobalScripts !== false) {
```

Comment:
is it our intention that the user could set `outputTarget.includeGlobalScripts` to `null` and have this condition evaluate to true? Seems off to me...

(just because `null !== false` evaluates to `true`)

we don't have to change that in this PR but just calling it out because I noticed it

Reply:
That does seem off. I'm hesitant to change it just because that was the logic already in place, but I can make note of that and take a look outside the context of this PR (this is going into v3 anyway)